azurerm/config.go

```diff
@@ -300,6 +303,13 @@ func getAuthorizationToken(c *authentication.Config, oauthConfig *adal.OAuthConf
 	}
 
 	spt, err := adal.NewServicePrincipalTokenFromManualToken(*oauthConfig, c.ClientID, endpoint, *c.AccessToken)
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = spt.Refresh()
```

Comment:
this is a great spot - so that we can ship this fix faster I've split this out into #1544 (and credited you in that PR) - I hope you don't mind